rpc/rpcserver/server.go

```diff
@@ -299,6 +300,25 @@ func (s *walletServer) Balance(ctx context.Context, req *pb.BalanceRequest) (
 	return resp, nil
 }
 
+func (s *walletServer) ReceivedForAddress(ctx context.Context, req *pb.ReceivedForAddressRequest) (
+	*pb.ReceivedForAddressResponse, error) {
+	address, err := btcutil.DecodeAddress(req.Address, &chaincfg.MainNetParams)
+	if err != nil {
+		return nil, translateError(err)
+	}
+
+	reqConfs := req.RequiredConfirmations
+	balance, err := s.wallet.TotalReceivedForAddr(address, reqConfs)
```

Comment:
total received is not the "balance".  The total received is never decremented when the output is spent, and the value can even exceed the max bitcoin amount because it is a sum of all received amounts.